be/src/cloud/cloud_base_compaction.cpp

```diff
@@ -49,7 +49,10 @@ CloudBaseCompaction::~CloudBaseCompaction() = default;
 
 Status CloudBaseCompaction::prepare_compact() {
     if (_tablet->tablet_state() != TABLET_RUNNING) {
-        return Status::InternalError("invalid tablet state. tablet_id={}", _tablet->tablet_id());
+        Status res = Status::InternalError("invalid tablet state. tablet_id={}", _tablet->tablet_id());
```

Comment:
Use defer samentic to avoid duplicated code, e.g.
```
Defer  defer_set_st([&failed, &st] {
    if (!failed) return;
    cloud_tablet()->set_last_full_compaction_failure_time(UnixMillis());
    cloud_tablet()->set_last_full_compaction_status(st.to_string());
});
```

Reply:
done